src/server/pkg/testutil/env.go

```diff
@@ -81,6 +82,12 @@ func WithEtcdEnv(cb func(*EtcdEnv) error) (err error) {
 		return err
 	}
 
+	// NOTE: this is changing a GLOBAL variable in etcd. This function should not
+	// be run in the same process as production code where this may affect
+	// performance (but there should be little risk of that as this is only for
+	// test code).
+	etcdwal.SegmentSizeBytes = 1 * 1000 * 1000 // 1 MB
```

Comment:
This was to fix in-process testing resulting in very large data directories (etcd was allocating something like 128 MB of WAL files for each test, this lowers it to 2 MB).